### PR TITLE
fix(router) old_route has wrong assignment order

### PR DESCRIPTION
**Pull request by chronolaw**
### Summary

The assignment of `old_route` is wrong, it should happen before `old_routes[route_id] = route`.

And in the previous code, the rebuild is actually a complete build, not an incremental build.

### CODE REVIEW

File: kong/router/atc_compat.lua [hunk 1]
```diff
@@ -517,6 +520,8 @@ local function new_from_previous(routes, is_traditional_compatible, old_router)
       inst:remove_matcher(id)
       old_routes[id] = nil
       old_services[id] = nil
+
+      ngx_log(ngx_DEBUG, "route(", route_id, ") is removed")
```
Comment (chronolaw) [inline kong/router/atc_compat.lua:524]: `route_id` should be `id`

File: spec/01-unit/08-router_spec.lua [hunk 2]
```diff
@@ -4251,3 +4251,96 @@ describe("[both regex and prefix with regex_priority]", function()
 
 end)
 
+
+describe("incremental rebuild", function()
```
Comment (chronolaw) [inline spec/01-unit/08-router_spec.lua:4255]: Could we move these code to line 1673, which tests `flavor ~= "traditional"`.

Comment (chronolaw) [reply spec/01-unit/08-router_spec.lua:4255]: @dndx , Could you take a look at these cases, Do you have better idea?

Comment (dndx) [reply spec/01-unit/08-router_spec.lua]: Moved.

**Review by dndx — APPROVED** [review]

**Pull request merged**